src/libs/Animals/TFishSchools.cpp

```diff
@@ -42,37 +38,37 @@ void TFishSchools::Init()
 
     renderService = static_cast<VDX9RENDER *>(core.CreateService("dx9render"));
     if (!renderService)
+    {
         throw std::runtime_error("!FishSchools: No service 'dx9render'");
+    }       
 
-    sea = static_cast<SEA_BASE *>(EntityManager::GetEntityPointer(EntityManager::GetEntityId("sea")));
+    sea = static_cast<SEA_BASE*>(EntityManager::GetEntityPointer(EntityManager::GetEntityId("sea")));
     if (!sea)
     {
         enabled = false;
         return;
     }
 
-    for (auto i = 0; i < fishSchoolsCount; i++)
+    for (auto& fishSchool : fishSchools)
     {
-        fishSchools[i] = new TFishSchool();
-        fishSchools[i]->Initialize(CVECTOR(0.0f, 0.0f, 0.0f), maxDistance);
-        fishSchools[i]->depth = 0.05f;
-        fishSchools[i]->time = 0;
-        fishSchools[i]->timeDivider = randUpper(100.0f);
-        fishSchools[i]->amplitude = rand(0.01f);
-        AddDeflector(fishSchools[i]);
+        fishSchool = std::make_unique<TFishSchool>();
+        fishSchool->Initialize(CVECTOR(0.0f, 0.0f, 0.0f), maxDistance);
+        fishSchool->depth = 0.05f;
+        fishSchool->time = 0;
+        fishSchool->timeDivider = randUpper(100.0f);
+        fishSchool->amplitude = rand(0.01f);
+        AddDeflector(move(fishSchool));
```

Comment:
You are moving elements out of a container that later iterates over the container (TFishSchools::ProcessMessage). I don't know if this is correct behaviour. Consider taking raw pointers in AddDeflector and AddAttractor functions and pass fishSchool.get() to them